### DISCUSSION TITLE
Cannot compile the mutation

### DISCUSSION THREAD

**Discussion by pnkapadia64** [MichalLytek/type-graphql#989] [Q&A]
I'm trying to work out a mutation to remove a favourite game corresponding to a user's email address. The input is pretty straightforward, two strings. I keep getting the error 
`NoExplicitTypeError: Unable to infer GraphQL type from TypeScript reflection system. You need to provide explicit type for 'removeUserFavouriteGameMutation' of 'FavouriteGamesResolver' class.`
The error is same even if the input doesn't extend an object (here FavouriteUserGame) or if I use an abstract class. Not sure what is causing it.

```
@ObjectType()
class FavouriteUserGame {
  @Field()
  email: string;
  @Field()
  gameSlug: string;
}

@InputType()
class FavouriteGameInput implements Partial<FavouriteUserGame> {
  @Field()
  email: string;
  @Field()
  gameSlug: string;
}

@Resolver()
export class FavouriteGamesResolver {
  @Mutation()
  @Authorized()
  async removeUserFavouriteGameMutation(@Args() { email, gameSlug }: FavouriteGameInput, @Ctx('dataSources') dataSources: UPDataSources) {
    // mutation
    return;
  }
}
```

**Comment by MichalLytek** [comment]
`@Mutation(type => SclaraOrObjectTypeClassYouReturn)`

If you want to return `undefined` or `void`, in GraphQL you should `return true;` and use `Boolean` scalar type.
